StudyConnect.API/Controllers/Groups/GroupCommentController.cs

```diff
@@ -0,0 +1,195 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Identity.Web;
+using StudyConnect.API.Dtos.Requests.Group;
+using StudyConnect.API.Dtos.Responses.Group;
+using StudyConnect.Core.Interfaces;
+using StudyConnect.Core.Models;
+using static StudyConnect.API.Extensions.MappingExtensions;
+using static StudyConnect.Core.Common.ErrorMessages;
+
+namespace StudyConnect.API.Controllers.Groups;
+
+/// <summary>
+/// Controller for managing the comments.
+/// Provides endpoints to create, retrieve, update, and delete comments.
+/// </summary>
+[ApiController]
+public class GroupCommentController : BaseController
+{
+    /// <summary>
+    /// The comment repository for data operations.
+    /// </summary>
+    protected readonly IGroupCommentRepository _commentRepository;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="GroupCommentController"/> class.
+    /// </summary>
+    /// <param name="commentRepository">The repository used to manage comment data.</param>
+    public GroupCommentController(IGroupCommentRepository commentRepository)
+    {
+        _commentRepository = commentRepository;
+    }
+
+    /// <summary>
+    /// Creates a new comment.
+    /// </summary>
+    /// <param name="gid">The unique identifier of the post.</param>
+    /// <param name="pid"></param>
```

Comment:
Nitpick but the comments here are wrong.
```suggestion
    /// <param name="gid">The unique identifier of the group.</param>
    /// <param name="pid">The unique identifier of the post.</param>
```